controllers/machinehealthcheck_controller.go

```diff
@@ -252,12 +254,57 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	for _, t := range unhealthy {
 		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
 
-		if annotations.IsPaused(cluster, t.Machine) {
-			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
+		if m.Spec.RemediationTemplate != nil {
+			// If external remediation request already exists,
+			// return early
+			if r.externalRemediationRequestExists(ctx, m, t) {
+				logger.Info("External Remediation Request for", t.Machine.Name, " exists")
+				return ctrl.Result{}, nil
+			}
```

Comment:
Can we inline the code in `externalRemediationRequestExists` and check the error message that `.Get` could return instead of ignoring it?

If there is a temporary issue connecting to the API server, we should make sure to return an error right away instead of trying to create a new resource from the template.